lab_6_pipeline/pipeline.py

```diff
@@ -18,22 +45,58 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self._storage = {}
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._validate_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        #do not use os, pathlib only
+        #glob can be used
+        texts = list(self.path_to_raw_txt_data.glob('*.txt'))
+        meta_info = list(self.path_to_raw_txt_data.glob('*.json'))
+        right_texts = sorted([int(re.match(r'\d+', i.name).group()) for i in texts])
+
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError
+
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError
+        if len(right_texts) != len(meta_info):
+            raise InconsistentDatasetError
+
+
+        num_of_article = 0
+        for i in right_texts:
+            if i-num_of_article != 1:
+                raise InconsistentDatasetError
+            with open(i, 'r', encoding= 'utf-8') as f:
+                content = f.read()
```

Comment:
it is not necessary to open file to know its size.

check size using Pathlib. https://docs.python.org/3/library/pathlib.html#pathlib.Path.stat